src/_pytest/python.py

```diff
@@ -192,6 +194,13 @@ def pytest_pyfunc_call(pyfuncitem: "Function") -> Optional[object]:
     result = testfunction(**testargs)
     if hasattr(result, "__await__") or hasattr(result, "__aiter__"):
         async_warn_and_skip(pyfuncitem.nodeid)
+    elif result is not None:
+        warnings.warn(
+            PytestReturnNotNoneWarning(
+                f"Expected None, but the test returned {result!r}, which will be an error in a "
+                "future version of Pytest.  Did you mean to use `assert` instead of `return`?"
```

Comment:
nitpick:

```suggestion
                "future version of pytest.  Did you mean to use `assert` instead of `return`?"
```

see a [recent Twitter thread](https://twitter.com/nicoddemus/status/1524499445752205314) :wink: